packages/core/src/modules/credentials/formats/jsonld/JsonLdCredentialFormatService.ts

```diff
@@ -330,41 +315,49 @@ export class JsonLdCredentialFormatService implements CredentialFormatService<Js
     })
   }
 
-  private compareCredentialSubject(
+  private verifyReceivedCredentialMatchesRequest(
     credential: W3cVerifiableCredential,
-    requestAsJson: JsonLdSignCredentialFormat
+    request: JsonLdFormatDataCredentialDetail
   ): void {
-    const request = JsonTransformer.fromJSON(requestAsJson.credential, W3cCredential)
+    // const request = JsonTransformer.fromJSON(requestAsJson.credential, W3cCredential)
 
-    if (!objectEquality(credential.credentialSubject, request.credentialSubject)) {
-      throw new AriesFrameworkError('Received credential subject does not match subject from credential request')
+    const jsonCredential = JsonTransformer.toJSON(credential)
+    delete jsonCredential.proof
```

Comment:
Do we know if `toJSON` does a shallow or deep copy? Deleting the proof field with a shallow copy would be problematic.